### PR TITLE
ObjectProperties parent property fix

### LINKED ISSUE #254: Accessing ObjectProperties `parent` raises AttributeError
The `object_properties` module does not import `cybox.core`. 

```
>>> from cybox.objects.address_object import Address
>>> addr = Address()
>>> addr.parent.id_
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/bworrell/projects/python-cybox/cybox/common/object_properties.py", line 90, in parent
    self._parent = cybox.core.Object(self)
AttributeError: 'module' object has no attribute 'core'
>>> cybox.__version__
'2.1.0.12.dev0'
```

### PR DESCRIPTION
**Pull request by bworrell**
This PR fixes #254, by adding `cybox.core` imports to the `ObjectProperties.parent` property. Importing it at the top of the module introduces some cyclic imports, so I went for this approach.

Adding a test for this was pretty difficult because the issue seems to resolve itself if `cybox.core.Object` is imported at some point before interacting with `parent`. To create a test, I made an `@isolate` decorator that sorta resets the `sys.modules` and `globals()` to a quasi-initial state and resets them after the  decorated function returns (or an exception is raised). 

I played around with the `--with-isolation` option in `nose`, but it didn't behave the way I expected, and opted for making `@isolate` and `isolation()`. I'm fine with removing them if it's a bit too much "magic" :)

### CODE REVIEW

**Comment by gtback** [comment]
I'd rather look at getting the Nose `isolation` to work. Less custom code for us to maintain. Otherwise, I'd like to migrate most of `cybox/test/__init__.py` to mixbox (a `testing` module or similar).

I'm a bit torn between putting the import in the middle of the function vs. at the bottom of the file. There are pros and cons to each. I'm probably OK with this as it is.

File: cybox/test/__init__.py [hunk 1]
```diff
@@ -14,6 +17,89 @@
...
+
+    Any code which leverages this context manager will need to re-import all
+    dependencies within.
+    """
+    def del_sys_modules(to_del):
+        for mod in to_del:
+            del sys.modules[mod]
+
+    def del_globals(to_del):
+        for g in globals_to_del:
```
Comment (gtback) [inline cybox/test/__init__.py]: Should this be just `to_del`?

File: cybox/test/__init__.py [hunk 2]
```diff
@@ -14,6 +17,89 @@
...
+    """Context manager that attempts to reset the sys.modules and globals()
+    to its initial, pre-`import cybox` state.
+
+    This helps eliminate issues where tests pass because of residuals caused
+    by previous imports.
+
+    Any code which leverages this context manager will need to re-import all
+    dependencies within.
+    """
+    def del_sys_modules(to_del):
```
Comment (gtback) [inline cybox/test/__init__.py]: Can we move these functions to the top level? I don't like having functions inside functions unless they are necessary for a closure, decorator, or similar.

File: cybox/test/__init__.py (hunk 1 above)
Comment (bworrell) [inline cybox/test/__init__.py]: Haha, yep! It's actually kinda dumb luck that it even works at all. Thanks Python scoping rules!

File: cybox/test/__init__.py (hunk 2 above)
Comment (bworrell) [inline cybox/test/__init__.py]: Definitely! I probably use nested functions more than I should :-\ Aside from the closure/decorator stuff, I tend to use them when a large function benefits from being broken up into blocks of repeated code, but the resulting sub-functions don't make much sense or have value outside of the parent context (nothing else calls them). I definitely struggle with the decision between nesting and breaking out a function in those cases!

**Comment by gtback** [comment]
So I don't lose the link: [nose isolation plugin](https://nose.readthedocs.org/en/latest/plugins/isolate.html)

**Comment by gtback** [comment]
In interesting news, I have the `import cybox.core` changes in a stash from 2 months ago. I probably never commited because I didn't have a good failing prior test :wink:

**Comment by gtback** [comment]
Oh, sorry, @bworrell . Just noticed you pushed new commits this. I just made the changes directly in master.

**Pull request closed without merging**